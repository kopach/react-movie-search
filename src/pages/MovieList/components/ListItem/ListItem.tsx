import React from 'react'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Card from 'react-bootstrap/Card'
import ListGroup from 'react-bootstrap/ListGroup'

const ListItem: React.FC<IListItemProps> = (props): JSX.Element => (
  <ListGroup.Item>
    <Row>
      <Col xs sm={12} md={5} lg={4}>
        <Card>
          <Card.Img src={props.posterURL} />
        </Card>
      </Col>
      <Col xs sm="12" md="7" lg="8">
        <Card>
          <Card.Body>
            <Card.Title>{props.movieTitle}</Card.Title>
            <Card.Text>{props.children}</Card.Text>
          </Card.Body>
        </Card>
      </Col>
    </Row>
  </ListGroup.Item>
)

export { ListItem }

export interface IListItemProps {
  posterURL: string
  movieTitle: string
  children: React.ReactNode
}
