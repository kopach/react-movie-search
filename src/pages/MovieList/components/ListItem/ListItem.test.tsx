import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { IListItemProps, ListItem } from './ListItem'

describe('<ListItem />', () => {
  let component: ReturnType<typeof shallow>

  const listItemPropsMock: IListItemProps = {
    posterURL: 'someURL',
    movieTitle: 'some title',
    children: 'some text',
  }

  beforeEach(() => {
    component = shallow(<ListItem {...listItemPropsMock}> </ListItem>)
  })

  it('should mount', () => {
    expect(component.length).toBe(1)
  })

  it('renders as expected', () => {
    const tree = renderer.create(<ListItem {...listItemPropsMock}> </ListItem>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
