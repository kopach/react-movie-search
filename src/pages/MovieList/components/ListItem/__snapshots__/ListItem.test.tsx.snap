// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders as expected 1`] = `
<div
  className="list-group-item"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-lg-4 col-md-5 col-sm-12 col"
    >
      <div
        className="card"
      >
        <img
          className="card-img"
          src="someURL"
        />
      </div>
    </div>
    <div
      className="col-lg-8 col-md-7 col-sm-12 col"
    >
      <div
        className="card"
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            some title
          </div>
          <p
            className="card-text"
          >
             
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
